code/controllers/subsystem/jobs.dm

```diff
@@ -392,7 +392,32 @@ var/datum/subsystem/job/SSjob
 			H.loc = S.loc
 
 	if(H.mind)
-		H.mind.assigned_role = rank
+		var/datum/mind/H_mind = H.mind
+		H_mind.assigned_role = rank
+
+		if(H.voiceprint && (!joined_late || world.time - round_start_time <= config.latejoin_identity_retcon))
```

Comment:
Where, the `world.time - round_start_time <= config.latejoin_identity_retcon`?
ATM it's H.voiceprint is true AND (they aren't a latejoin OR the latejoin backfill time hasn't expired)